pyccel/codegen/printing/cwrappercode.py

```diff
@@ -302,6 +381,11 @@ def _print_FunctionDef(self, expr):
 
         # Call PyBuildNode
         wrapper_body.append(AliasAssign(wrapper_results[0],PyBuildValueNode(res_args)))
+
+        # Call free function for python type
+        wrapper_body += [FunctionCall(Py_DECREF, [i]) for i in self._to_free_PyObject_list]
```

Comment:
What object is being freed here?

Reply:
When returning a complex number  we use function PyComplex_FromDouble https://docs.python.org/2/c-api/complex.html#c.PyComplex_FromDoubles to create the python type complex .
Then we return  a pytObject  with PyBuild_value  that create a new pyobject from given arg . so the variable that we created with PyComplex_FromDouble is not freed  
```
@types('complex')
def f(c):
    return c


PyObject *f_wrapper(PyObject *self, PyObject *args, PyObject *kwargs)
{
    double complex c;
    double complex Out_0001;
    PyObject *c_tmp;
    PyObject *Out_0001_tmp;
    PyObject *result;
    static char *kwlist[] = {
        "c",
        NULL
    };

    if (!PyArg_ParseTupleAndKeywords(args, kwargs, "O", kwlist, &c_tmp))
    {
        return NULL;
    }
    c = pycomplex_to_complex(c_tmp);
    Out_0001 = f(c);
    Out_0001_tmp = complex_to_pycomplex(Out_0001);   // Pycomplex created with  PyComplex_FromDouble 
    result = Py_BuildValue("O", Out_0001_tmp);  // new Pyobject created from Out_0001_tmp 
    Py_DECREF(Out_0001_tmp);   // should free Out_0001_tmp 
    return result; 
}

```

Reply:
Ok that makes sense :+1